tests/Unit/DataStructures/Test_TensorMagnitude.cpp

```diff
@@ -7,90 +7,92 @@
+  const tnsr::i<DataVector, 1, Frame::Grid> negative_one_d_covector{
+      {{minus_three}}};
+  const DataVector magnitude_negative_one_d_covector =
+      magnitude(negative_one_d_covector);
+  for (size_t s = 0; s < npts; ++s) {
+    CHECK(magnitude_negative_one_d_covector[s] == 3.0);
+  }
+  const tnsr::A<DataVector, 1, Frame::Grid> one_d_vector{{{minus_three, four}}};
+  const DataVector magnitude_one_d_vector = magnitude(one_d_vector);
   for (size_t s = 0; s < npts; ++s) {
     CHECK(magnitude_one_d_vector[s] == 5.0);
   }
 
-  tnsr::I<DataVector, 2, Frame::Grid> two_d_vector;
-  two_d_vector.get<0>() = minus_five;
-  two_d_vector.get<1>() = twelve;
-  DataVector magnitude_two_d_vector = magnitude(two_d_vector);
+  const tnsr::I<DataVector, 2, Frame::Grid> two_d_vector{
+      {{minus_five, twelve}}};
+  const DataVector magnitude_two_d_vector = magnitude(two_d_vector);
   for (size_t s = 0; s < npts; ++s) {
     CHECK(magnitude_two_d_vector[s] == 13.0);
   }
 
-  tnsr::i<DataVector, 3, Frame::Grid> three_d_covector;
-  three_d_covector.get<0>() = minus_three;
-  three_d_covector.get<1>() = twelve;
-  three_d_covector.get<2>() = four;
-  DataVector magnitude_three_d_covector = magnitude(three_d_covector);
+  const tnsr::i<DataVector, 3, Frame::Grid> three_d_covector{
+      {{minus_three, twelve, four}}};
+  const DataVector magnitude_three_d_covector = magnitude(three_d_covector);
   for (size_t s = 0; s < npts; ++s) {
     CHECK(magnitude_three_d_covector[s] == 13.0);
   }
 
   // 5D example
-  tnsr::a<DataVector, 4, Frame::Grid> five_d_covector;
-  five_d_covector.get<0>() = two;
-  five_d_covector.get<1>() = twelve;
-  five_d_covector.get<2>() = four;
-  five_d_covector.get<3>() = one;
-  five_d_covector.get<4>() = two;
-  DataVector magnitude_five_d_covector = magnitude(five_d_covector);
+  const tnsr::a<DataVector, 4, Frame::Grid> five_d_covector{
+      {{two, twelve, four, one, two}}};
+  const DataVector magnitude_five_d_covector = magnitude(five_d_covector);
   for (size_t s = 0; s < npts; ++s) {
     CHECK(magnitude_five_d_covector[s] == 13.0);
   }
 }
 
 TEST_CASE("Unit.DataStructures.Tensor.Magnitude", "[DataStructures][Unit]") {
-  const size_t npts = 8;
-  DataVector one(npts, 1.0);
-  DataVector two(npts, 2.0);
-  DataVector minus_three(npts, -3.0);
-  DataVector four(npts, 4.0);
-  DataVector minus_five(npts, -5.0);
-  DataVector twelve(npts, 12.0);
-  DataVector thirteen(npts, 13.0);
+  const size_t npts = 2;
+  const DataVector one(npts, 1.0);
+  const DataVector two(npts, 2.0);
+  const DataVector minus_three(npts, -3.0);
+  const DataVector four(npts, 4.0);
+  const DataVector minus_five(npts, -5.0);
+  const DataVector twelve(npts, 12.0);
+  const DataVector thirteen(npts, 13.0);
+
+  const tnsr::i<DataVector, 1, Frame::Grid> one_d_covector{{{two}}};
+  const tnsr::II<DataVector, 1, Frame::Grid> inv_h = [&four]() {
+    tnsr::II<DataVector, 1, Frame::Grid> tensor;
+    tensor.template get<0, 0>() = four;
```

Comment:
Yes, the `[]` is a capture list, which you can think of as the arguments to the constructor. The `&` before means it's being captured by reference, so no copy is done but if the lambda were returned from a function then there would be undefined behavior. An equally valid thing to do would be to put it in the `()` which are the arguments to the call operator `operator()` for the class.